velox/common/base/SimdUtil-inl.h

```diff
@@ -1436,4 +1436,146 @@ inline bool memEqualUnsafe(const void* x, const void* y, int32_t size) {
   return true;
 }
 
+namespace detail {
+
+/// NOTE: SSE_4_2`s the performance of simdStrStr is a little slower than
+/// std::find in first-char-unmatch(read only one char per match.) Use AVX2 the
+/// performance will be better than std::find in that case.
+#if XSIMD_WITH_SSE4_2
+using CharVector = xsimd::batch<uint8_t, xsimd::sse4_2>;
+#elif XSIMD_WITH_NEON
+using CharVector = xsimd::batch<uint8_t, xsimd::neon>;
+#endif
+
+FOLLY_ALWAYS_INLINE bool pageSafe(const void* const ptr, size_t length) {
+  static const int kPageSize = sysconf(_SC_PAGESIZE);
```

Comment:
I don't think `static` would be faster than global in normal cases.  Can you check if compiler is inlining this value in this case?  If so it is good; otherwise we should still try global.

Reply:
> I don't think `static` would be faster than global in normal cases. Can you check if compiler is inlining this value in this case? If so it is good; otherwise we should still try global.

![image](https://github.com/user-attachments/assets/bba073c8-8d7d-461f-9c74-e239e8592493)


I think the  kPageSize could be inline ,maybe I am wrong , could you have a look at this?

Reply:
No it's not inlined, it's checking `guard_variable_for_pageSafe` the first time you use it and calculate the value and store it in `kPageSize`, and load it in `eax` every time you used it.  I think use a global variable should save us the check for `guard_variable_for_pageSafe`.  You can try it to see what assembly it generates.

Reply:
![image](https://github.com/user-attachments/assets/79a3fe05-2a7a-4d01-9870-091d1f5470c8)

yeah, you are right, thanks for explaining this~